python_modules/dagster/dagster/_core/instance/__init__.py

```diff
@@ -1778,6 +1779,52 @@ def get_event_records(
         """
         return self._event_storage.get_event_records(event_records_filter, limit, ascending)
 
+    @public
+    @traced
+    def get_status_by_partition(
+        self, asset_key: AssetKey, partitions_def: PartitionsDefinition, partition_keys: List[str]
+    ) -> Dict:
```

Comment:
Nitpick: the return type should be parameterized, and generally we prefer `Mapping` instead of `Dict`, because it's immutable and thus gives us freedom to cache values without worrying that the caller might mutate them.

So `Mapping[str, PartitionStatus]`.